### PR TITLE
HOLD: Add initial celery setup

### PR DESCRIPTION
**Pull request by mbeacom**
##### Description
The goal of this PR is to implement Celery using Redis as the broker and backend and add the first scheduled task: `sync_profile_data`.

Refs: #520

##### Checklist
- [x] linter status: 100% pass
- [x] changes don't break existing behavior
- [x] commit message follows [commit guidelines](https://github.com/gitcoinco/web/blob/master/docs/CONTRIBUTING.md#step-4-commit)

##### Affected core subsystem(s)
backend

### CODE REVIEW

File: app/dashboard/tasks.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        org.last_sync_date > timezone.now() - timezone.timedelta(weeks=1)
+    except Exception:
+        needs_refresh = True
+
+    return needs_refresh
+
+
+@app.task
+def sync_profile_data(force_refresh=False, iteration_sleep=False):
+    """Sync all handle data of current bounties."""
```
Comment (owocki) [inline app/dashboard/tasks.py]: i wonder waht the best pratice is here: should each handle be its own task? 🤔 or do we want one large long running task for each handle?

**Comment by owocki** [comment]
i wonder if we want to run this all on the prod server or if we want to spin up another server for running background tasks upon this deployment?

**Comment by owocki** [comment]
didnt test, but code looks sane to me. 

do we want to add some stuff to the documentation about how to manage / inspect the celery queues?

**Comment by mbeacom** [comment]
@owocki Yeah, I'll update the documentation to address the added containers, configuration, etc.

As for deployment, I was planning on using AWS ElastiCache for Redis and either a secondary box running celery worker and beat or an instance(s) using ecs or k8 to run worker and beat as individual containers.
Personally, I'm leaning towards the containerized route.  That'll free us up to simply scale out the cluster when we redeploy the web app as a container.

**Comment by owocki** [comment]
> As for deployment, I was planning on using AWS ElastiCache for Redis and either a secondary box running celery worker and beat or an instance(s) using ecs or k8 to run worker and beat as individual containers.

👍 

> Personally, I'm leaning towards the containerized route. That'll free us up to simply scale out the cluster when we redeploy the web app as a container.

Okie dokie.. We should deploy not-on-a-friday sometime so we can make sure everything is working correctly once its deployed!

**Comment by mbeacom** [comment]
@owocki We can deploy this as early as tomorrow.  Sound good?  Also, should I include or raise another PR for an IPFS service container?  That'll allow us to remove the ipfs subproc calls and just point to the hostname of the IPFS container.  With those offloaded, I'd think we could start deploying the web service on ECS/K8 or even AWS Lambda via Zappa or Chalice. Additionally, we don't *have to* eliminate local ipfs daemons, executing it via dumb-init as a background process when necessary, but I'd like to decouple the services and try to limit them to one service per container.

**Comment by owocki** [comment]
> We can deploy this as early as tomorrow. Sound good?

sure.  maybe start with stage first?

> Also, should I include or raise another PR for an IPFS service container?

makes sense to me.

**Comment by mkosowsk** [comment]
@mbeacom think this deployment could be an interesting topic for architecture chat on Thursday at 12 noon if you can make it :)

**Comment by mbeacom** [comment]
@mkosowsk Sounds good!

File: docker-compose.yml [hunk 2]
```diff
@@ -41,8 +60,55 @@ services:
...
+      - app/app/.env
+    environment:
+      - SERVICE_TYPE=beat
+      - CELERY_APP=app
+    volumes:
+      - .:/code
+    depends_on:
+      - celery_worker
+    command: ["/code/bin/celery/worker.bash"]
+    restart: unless-stopped
```
[the remark this answers is not in the document]
Comment (mbeacom) [reply docker-compose.yml]: Stickler, doing work.

**Comment by abitrolly** [comment]
This needs a rebase.

**Comment by thelostone-mc** [comment]
https://github.com/gitcoinco/web/issues/520#issuecomment-465271848

**Pull request closed without merging**